gym/spaces/dict.py

```diff
@@ -75,66 +89,68 @@ def __init__(
             seed: Optionally, you can use this argument to seed the RNGs of the spaces that make up the :class:`Dict` space.
             **spaces_kwargs: If ``spaces`` is ``None``, you need to pass the constituent spaces as keyword arguments, as described above.
         """
-        assert (spaces is None) or (
-            not spaces_kwargs
-        ), "Use either Dict(spaces=dict(...)) or Dict(foo=x, bar=z)"
-
-        if spaces is None:
-            spaces = spaces_kwargs
+        # Convert the spaces into an OrderedDict
         if isinstance(spaces, dict) and not isinstance(spaces, OrderedDict):
             try:
                 spaces = OrderedDict(sorted(spaces.items()))
             except TypeError:  # raise when sort by different types of keys
                 spaces = OrderedDict(spaces.items())
-        if isinstance(spaces, Sequence):
+        elif isinstance(spaces, Sequence):
             spaces = OrderedDict(spaces)
```

Comment:
I see now that in the existing code we're using OrderedDict, which imo we should change to just being a normal dict. It doesn't have to be done in this PR, but we shouldn't get additionally "tied" to this approach

Reply:
Agreed, the problem is that python 3.6 dicts are not ordered. We could ignore this or make it a gymnasium change?